app/component/search/search-modal.cjsx

```diff
@@ -13,22 +15,9 @@ class SearchModal extends React.Component
     getStore: React.PropTypes.func.isRequired
     executeAction: React.PropTypes.func.isRequired
 
-  componentWillMount: =>
-    @context.getStore('SearchStore').addChangeListener @onSearchChange
-
-  componentWillUnmount: =>
-    @context.getStore('SearchStore').removeChangeListener @onSearchChange
-
-  onSearchChange: (a) =>
-    position = @context.getStore('SearchStore').getPosition()
-    @setState("value": position?.address || "")
-
-  closeModal: () =>
-    @context.executeAction SearchActions.closeSearch
-
   render: =>
     style = {}
-    if @context.getStore('SearchStore').isModalOpen() == false
+    if @props.modalIsOpen == false
```

Comment:
There might be reason why this is done like this. However, I'm wondering why are we rendering the modal in the first place when it is closed.

Why not render it only when "modalIsOpen" ?